libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -338,14 +338,23 @@ string IRGenerator::generateModifier(
 		t("retParams", joinHumanReadable(retParams));
 		t("assignRetParams", assignRetParams);
 
-		solAssert(*_modifierInvocation.name().annotation().requiredLookup == VirtualLookup::Virtual, "");
-
-		ModifierDefinition const& modifier = dynamic_cast<ModifierDefinition const&>(
+		ModifierDefinition const* modifier = &dynamic_cast<ModifierDefinition const&>(
 			*_modifierInvocation.name().annotation().referencedDeclaration
```

Comment:
Maybe better to cast to pointer directly?

```suggestion
		auto const* modifier = dynamic_cast<ModifierDefinition const*>(
 			_modifierInvocation.name().annotation().referencedDeclaration
```

Reply:
If we do that, I have to add another solAssert, but I can do that.